lib/buildingsync/model_articulation/site.rb

```diff
@@ -52,12 +57,28 @@ def initialize(build_element, ns)
       @state_name = nil
       @latitude = nil
       @longitude = nil
+      @street_address = nil
+      @postal_code = nil
+
+
       # TM: just use the XML snippet to search for the buildings on the site
       read_xml(build_element, ns)
     end
 
     # adding a site to the facility
     def read_xml(build_element, ns)
+      # first we check if the number of buildings is ok
+      number_of_buildings = 0
+      build_element.elements.each("#{ns}:Buildings/#{ns}:Building") do |buildings_element|
+        number_of_buildings += 1
+      end
+      if number_of_buildings == 0
+        OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Site.generate_baseline_osm', 'There is no building attached to this site in your BuildingSync file.')
+        raise 'Error: There is no building attached to this site in your BuildingSync file.'
+      elsif number_of_buildings > 1
+        OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Site.generate_baseline_osm', "There is more than one (#{number_of_buildings}) building attached to this site in your BuildingSync file.")
```

Comment:
Change to: `OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Site.read_xml`...

Reply:
done